From 02db52347ae3a2655abd36bea0098021cd9cecf3 Mon Sep 17 00:00:00 2001
From: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Date: Wed, 2 Oct 2024 17:14:16 +0200
Subject: [PATCH 1251/1695] dts: stm32: add SCMI HSI/HSE/MSI Kernel clocks for
 stm32mp23x

Adds CK_SCMI_HSI_KER_CK, CK_SCMI_HSE_KER_CK and CK_SCMI_MSI_KER_CK clocks

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 199bfa0fbec4f6f66610af8ab6fbcc6b9402f8cc dts: stm32: add SCMI HSI/HSE/MSI Kernel clocks for stm32mp23x]
Signed-off-by: Gabriel Fernandez <gabriel.fernandez@foss.st.com>
Change-Id: I55d8c87ff6d4c929ecf8b93fdd12a0e947e35c64
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/408609
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/arch/arm/dts/stm32mp231.dtsi | 42 +++++++++++++++++++++++++++++++
 1 file changed, 42 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp231.dtsi b/core/arch/arm/dts/stm32mp231.dtsi
index 746e88dd2..c0c45a3c8 100644
--- a/core/arch/arm/dts/stm32mp231.dtsi
+++ b/core/arch/arm/dts/stm32mp231.dtsi
@@ -1045,6 +1045,27 @@
 						clocks = <&rcc CK_KER_ETR>;
 						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
 					};
+
+					clock@7d {
+						reg = <CK_SCMI_HSI_KER_CK>;
+						domain-name = "hsi_ker_ck";
+						clocks = <&rcc HSI_KER_CK>;
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@7e {
+						reg = <CK_SCMI_HSE_KER_CK>;
+						domain-name = "hse_ker_ck";
+						clocks = <&rcc HSE_KER_CK>;
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@7f {
+						reg = <CK_SCMI_MSI_KER_CK>;
+						domain-name = "msi_ker_ck";
+						clocks = <&rcc MSI_KER_CK>;
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
 				};
 			};
 
@@ -2042,6 +2063,27 @@
 						clocks = <&rcc CK_KER_ETR>;
 						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
 					};
+
+					clock@7d {
+						reg = <CK_SCMI_HSI_KER_CK>;
+						domain-name = "hsi_ker_ck";
+						clocks = <&rcc HSI_KER_CK>;
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@7e {
+						reg = <CK_SCMI_HSE_KER_CK>;
+						domain-name = "hse_ker_ck";
+						clocks = <&rcc HSE_KER_CK>;
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
+
+					clock@7f {
+						reg = <CK_SCMI_MSI_KER_CK>;
+						domain-name = "msi_ker_ck";
+						clocks = <&rcc MSI_KER_CK>;
+						flags = <SCMI_CLOCK_DEFAULT_ENABLED>;
+					};
 				};
 			};
 
-- 
2.47.2

